Azure Services/Azure Monitor/Agents/Migration Tools/DCR Config Generator/WorkspaceConfigToDCRMigrationTool.ps1

```diff
@@ -609,18 +737,53 @@ function Get-DCRStream
         "WindowsPerformanceCounter" { $stream += "Microsoft-Perf"; Break }
         "LinuxPerformanceObject" { $stream += "Microsoft-Perf"; Break }
         "WindowsEvent" { $stream += "Microsoft-Event"; Break }
-        "LinuxSyslog" { $stream += "Microsoft-Syslog" }
+        "LinuxSyslog" { $stream += "Microsoft-Syslog"; Break }
+        "CustomLog" 
+        {
+            $workspaceDataSourceList = Get-AzOperationalInsightsDataSource -ResourceGroupName $ResourceGroupName -WorkspaceName $WorkspaceName -Kind $dataSourceType
+            foreach($dataSource in $workspaceDataSourceList)
+            {
+                if ($dataSource.Properties -ne $null -and $dataSource.Properties.customLogName -ne $null)
+                {
+                    $stream += "Custom-" + $dataSource.Properties.customLogName
+                }
+            }
+            Break
+        }
+
+        "IISLog" { $stream += "Microsoft-W3CIISLog" }
     }
 
     return $stream
 }
 
+function GetOrCreate-DataCollectionEndpoint
+{
+    param (
+        [Parameter(Mandatory=$true)][string] $SubscriptionId,
+        [Parameter(Mandatory=$true)][string] $ResourceGroupName,
+        [Parameter(Mandatory=$true)][string] $WorkspaceName
+    )
+    
+    $dceName = $SubscriptionId + "-dce"
+    # If DCE does not exist, it will create a new one. If it does exists, it will return the existing one
+    $dce = az monitor data-collection endpoint create --name $dceName --public-network-access "Enabled" --resource-group $ResourceGroupName
```

Comment:
Should public network access be enabled?

Reply:
Yes. Unless the customer wants to configure PLS (We can discuss that later)